### DISCUSSION TITLE
How to use the repeater component in a template?

### DISCUSSION THREAD

**Discussion by Drummond-Dev** [PascaleBeier/filament-pages#4] [Q&A] [answered]
I'm creating a template so that the user can create and manage a page that has a carousel..

I thought of using the repeater component to upload the carousel images, but to no avail.

giving the error: foreach() argument must be of type array|object, int given

How can I create a template that uses the repeater?

**Comment by PascaleBeier** [comment]
Hey, could you share what you got so far?

**Reply by Drummond-Dev** [reply]
Hi Pascale.

I managed to do what I wanted. It was my mistake when structuring the components. What I did was give the possibility to the user to upload images that will be presented in a carousel in the header of the page. Using Filament's repeater component.

**Reply by Drummond-Dev** [reply]
My question now is how to show the page (template) to the user.

Can I create a route to a controller and make the template display?

I'm sorry but this is my first time using Filament and its plugin

**Reply by PascaleBeier** [reply]
Right! I'm travelling right now, but there is a controller right here in the source code that you could use for reference for how to implement front routing

**Reply by Drummond-Dev** [reply, marked as answer]
oh I'm sorry.

I'm creating the components for Blade and then I'll look for it and if I have any doubts I'll get back to you, thanks
